Sources/NIOTransportServices/NIOTSNetworkEvents.swift

```diff
@@ -43,6 +43,20 @@ public enum NIOTSNetworkEvents {
         /// Create a new ``NIOTSNetworkEvents/BetterPathUnavailable`` event.
         public init(){ }
     }
+    
+    /// ``IsViable`` is triggered when the OS informs NIO that communication
+    /// with the remote endpoint is possible, indicating that the connection is viable.
+    public struct IsViable: NIOTSNetworkEvent {
```

Comment:
I know this follows the pattern for `BetterPathAvailable`/`BetterPathUnavailable`, but is there any reason to not have a single event struct, e.g. `ViabilityUpdate` with an `isViable` flag?

Reply:
I agree I can make the name change, I just wanted to follow the current the pattern. There are a handful of things I would like to update with NIOTS, but just picking a few things off at a time.  I also didn't write any tests because I didn't see any others for the other handlers, but maybe I missed them. Give me an hour in the morning and I will have an updated PR.

Reply:
Maybe @Lukasa has some insight into why there are separate events types for path availability?

Reply:
I kind of like the Idea of using an enum. What do you think?

```swift
private func viabilityUpdateHandler(viable: Bool) {
        self.pipeline.fireUserInboundEventTriggered(NIOTSNetworkEvents.ViablitiyUpdate(viable: viable ? .isViable : .isNotViable))
    }

    /// ``IsViable`` is triggered when the OS informs NIO that communication
    /// with the remote endpoint is possible, indicating that the connection is viable.
    public struct ViablitiyUpdates: NIOTSNetworkEvent {
        
        public let viability: Viability
        
        public enum Viablitiy {
            case .isViable, .isNotViable
        }
        
        /// Create a new ``NIOTSNetworkEvents/IsViable`` event.
        public init(viability: Viability){
            self.viability = viability
        }
    }
```

From my viewpoint this is cleaner and nicer to work with. Would this be an acceptable approach for NIOTransportServices? @Lukasa @glbrntt

Reply:
This is just a `Bool` with a fancy hat on 🤠 – I'd prefer to keep things simple here and stick with an `isViable` `Bool`.

Reply:
Good point. I think this is a bit better now.